### PR TITLE
[master][SAI-PTF]Split L3 subtests into individual tests

### PR DESCRIPTION
**Pull request by ravi861**
This PR addresses issues #1604 and #1599 

This PR updates following objects to address above issues

sairoute
saineighbor
sainexthop
sainexthopgroup

### CODE REVIEW

File: ptf/sairoute.py [hunk 1]
```diff
@@ -1050,10 +1089,20 @@ def gleanEndForwardTest(self):
             sai_thrift_remove_neighbor_entry(self.client, nbr_entry2)
```
Comment (richardyu-ms) [inline ptf/sairoute.py:1089]: local variable nbr_entry1 referenced before assignment

File: ptf/sainexthop.py [hunk 2]
```diff
@@ -677,22 +721,19 @@ def setUp(self):
...
             encap_ttl_mode=self.ttl_mode,
             decap_ttl_mode=self.ttl_mode,
             underlay_interface=self.urif_lb)
 
-    def runTest(self):
-        self.tunnelVniTest()
-
     def tearDown(self):
         sai_thrift_remove_tunnel(self.client, self.tunnel)
         sai_thrift_remove_tunnel_map_entry(self.client,
```
Comment (richardyu-ms) [inline ptf/sainexthop.py:739]: self.decap_tunnel_map_entry is not an integer

File: ptf/sairoute.py (hunk 1 above)
Comment (richardyu-ms) [reply ptf/sairoute.py:1089]: fixed in https://github.com/ravi861/SAI/pull/3 and https://github.com/ravi861/SAI/pull/2

File: ptf/sainexthop.py (hunk 2 above)
Comment (richardyu-ms) [reply ptf/sainexthop.py:739]: fixed in https://github.com/ravi861/SAI/pull/3 and https://github.com/ravi861/SAI/pull/2

**Review by richardyu-ms — APPROVED** [review]

File: ptf/sai_base_test.py [hunk 3]
```diff
@@ -203,9 +290,38 @@ def get_active_port_list(self):
...
+            port_up = True
+            if port_attr['oper_status'] != SAI_PORT_OPER_STATUS_UP:
+                port_up = False
+                for num_of_tries in range(retries):
+                    port_attr = sai_thrift_get_port_attribute(
+                        self.client, oid, oper_status=True)
+                    if port_attr['oper_status'] == SAI_PORT_OPER_STATUS_UP:
+                        port_up = True
+                        break
+                    time.sleep(5)
```
Comment (kcudnik) [inline ptf/sai_base_test.py:318]: @ravi861 please explain this sleeps if they are needed, they slow down out tests, waiting and retrying all time

Comment (ravi861) [reply ptf/sai_base_test.py:318]: Although the changes came in via this PR, this specific timeout was initiated by Richard during his testing with BRCM devices as the comment says above.
Typically, the test would not have to retry so many times if the links come up fast enough. An alternative to hard-coding 5 sec is to pass in a test param and make it configurable.

Comment (kcudnik) [reply ptf/sai_base_test.py:318]: in the DASH repo scenario where those tests are played, they are played gains p4/bmv2 switch, and those ports are never get up, at least what i observed notifications are not supported yet, so it waits 5x10 - 50 seconds to retry, and this is not the only tests, there are couple of them

Comment (kcudnik) [reply ptf/sai_base_test.py:318]: you can see the retrying here: https://github.com/sonic-net/DASH/actions/runs/6053205361/job/16428197918?pr=432#step:17:80

**Pull request merged**